beamer/agent/middleware.py

```diff
@@ -39,13 +57,14 @@ def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
             response = make_request(method, params)
             if _result_ok(response):
                 key = hex(response["result"].number), params[1]
-                cache[key] = response
+                _BLOCK_CACHE[chain_id][key] = response
```

Comment:
Why do we store the whole response, instead of just `response['result']`?

Reply:
don't we have to store the whole response in order to imitate the actual call?

Reply:
yes it is because we need to return the whole response from middleware